includes/embeds/class-amp-gallery-embed.php

```diff
@@ -128,7 +128,10 @@ public function shortcode( $attr ) {
 	 * @param array  $attributes Shortcode attributes.
 	 * @return string $html Markup for the gallery.
 	 */
-	public function override_gallery( $html, $attributes ) {
+	public function maybe_override_gallery( $html, $attributes ) {
+		if ( isset( $attributes['amp-carousel'] ) && false === filter_var( $attributes['amp-carousel'], FILTER_VALIDATE_BOOLEAN ) ) {
+			return $html;
```

Comment:
@westonruter Added editor support for making displaying gallery shortcode as `amp-carousel` optional by adding a shortcode attribute `amp-carousel`. The attribute is hidden from the user in editor view, and the Inspector Control is displayed only if `[gallery` is found in the shortcode text, seems to be working as expected.
![screen shot 2018-04-26 at 4 30 05 pm](https://user-images.githubusercontent.com/3294597/39308969-8aefad3a-496f-11e8-9d3c-48daf4ea1205.png)

Do you think this approach makes sense? Thoughts?

Reply:
Yes, I think that makes sense and it seems to work great.

At first I was thinking that carousel option should also be made available to the Gallery block in addition to the `gallery` shortcode, but maybe that's something that will be coming in Gutenberg as a separate block type in its own right.

Reply:
Do you think that just using `amp-carousel` attribute within post content when using Classic Editor for gallery shortcode could be a fix for #1065, too?

Reply:
Yes, it would, just thinking if this should be the official fix that closes #1065, too, or if an alternative approach would be needed for that.